### PR TITLE
[INFINITY-1366] Changed services to run as 'nobody' by default instead of 'root'

### PR DESCRIPTION
**Pull request by NimaVaziri**
Confirmed UID of files in tasks is 'nobody'

### CODE REVIEW

**Comment by benclarkwood** [comment]
Once the DC/OS bug is fixed, I will be curious to see if our tests pass running as `nobody` or not

**Comment by benclarkwood** [comment]
retest this please

**Comment by paegun** [comment]
@NimaVaziri, @benclarkwood this change only sets the default, but several of the `frameworks/$framework/src/main/dist/svc.yml` are not setting user at the scheduler-, let alone pod-level.

The following is a list of frameworks and how they differ from what I believe is the expectation, that the service allows for override of the default via install options as well as configuration edit:

| framework | scheduler-user | pod-user |
|------------|----------------|-----------|
| cassandra | ✅  | ❌ |
| dse | ❌ | ❌ |
| elastic | ✅  | ❌ |
| hdfs | ❌ | ❌ |
| helloworld | ✅ | ✅ |
| kafka | ✅ | ❌ |
| confluent | ✅ | ❌ |
|template | ✅ | ✅ |


*) Additionally, some of the frameworks use `{{SERVICE_USER}}` and `{{SERVICE_PRINCIPAL}}` while others use `{{FRAMEWORK_USER}}` and `{{FRAMEWORK_PRINCIPAL}}` . I believe while touching all of the same files that would be required to make a change to standardize on either SERVICE_ or FRAMEWORK_ , barring bike shedding, we should make the change to standardize on one, which afaik is SERVICE_ . TBH I can't determine if we should use FRAMEWORK_ or SERVICE_ as the count of each is near even and helloworld uses SERVICE_ while template uses FRAMEWORK_ .

**Comment by NimaVaziri** [comment]
@paegun yeah you're correct in that the user needs to be set at several more places to match expected behavior. The first commit was just to initialize the PR :)

Re: consistency issue - IMO it'd make more sense to standardize on `SERVICE_` since the docs/guide refer to them as services (in which case we should probably also rename the `frameworks` folder to `services`)

**Comment by NimaVaziri** [comment]
All the scheduler and pod users have been set now, meaning executors and schedulers running permissive and strict mode, with the exception of the scheduler in permissive mode: in this case the scheduler seems to be not able to be ran as `nobody` despite setting the service's user in the config and updating the scheduler code to provide this info in the right protobuf.

AFAICT [this protobuf](https://github.com/apache/mesos/blob/master/include/mesos/mesos.proto#L218) indicates that only the executors can run as the given `user` so strict mode probably has the default scheduler user hardwired to `nobody`.

**Comment by NimaVaziri** [comment]
retest please

**Comment by benclarkwood** [comment]
@NimaVaziri Can you please clean up conflicts? CI should be more stable now.

**Comment by NimaVaziri** [comment]
Yup merge resolution coming up

**Comment by benclarkwood** [comment]
retest this please

**Comment by NimaVaziri** [comment]
retest this please

File: frameworks/hdfs/src/main/dist/svc.yml [hunk 1]
```diff
@@ -1,9 +1,11 @@
 name: {{FRAMEWORK_NAME}}
 scheduler:
   principal: {{SERVICE_PRINCIPAL}}
+  user: {{TASKCFG_ALL_TASK_USER}}
```
Comment (NimaVaziri) [inline frameworks/hdfs/src/main/dist/svc.yml]: Ah alright, in that case I'll remove the wiring I setup in this PR to inject this into the FrameworkInfo protobuf then.

Comment (benclarkwood) [inline frameworks/hdfs/src/main/dist/svc.yml]: Oh... I didn't notice that. Before you do that, @nickbp @gabrielhartmann is that required to make strict work?

Comment (NimaVaziri) [inline frameworks/hdfs/src/main/dist/svc.yml]: AFAIK it's not required

**Comment by NimaVaziri** [comment]
Closed in favor of splitting this on other service-specific branches.

**Pull request closed without merging**